pkg/meta/tkv.go

```diff
@@ -1977,26 +1977,26 @@ func (m *kvMeta) doGetParents(ctx Context, inode Ino) map[Ino]int {
 	return ps
 }
 
-func (m *kvMeta) doSyncDirStat(ctx Context, ino Ino) (*dirStat, error) {
+func (m *kvMeta) doSyncDirStat(ctx Context, ino Ino) (*dirStat, syscall.Errno) {
 	if m.conf.ReadOnly {
 		return nil, syscall.EROFS
 	}
-	stat, err := m.calcDirStat(ctx, ino)
-	if err != nil {
-		return nil, err
+	stat, st := m.calcDirStat(ctx, ino)
+	if st != 0 {
+		return nil, st
 	}
-	err = m.client.txn(func(tx *kvTxn) error {
+	err := m.client.txn(func(tx *kvTxn) error {
+		if !tx.exist(m.inodeKey(ino)) {
```

Comment:
better :
```
a := tx.get(m.inodeKey(ino))
    if a == nil {
    return syscall.ENOENT
}
```
`exist` isn't a direct interface to the metadata engine.
`exist` is a confusing name. eg: `m.exist(m.entryKey(parent, name))` is wrong usage in tkv but right  in redis.
```
func (tx *tikvTxn) exist(prefix []byte) bool {
	it, err := tx.Iter(prefix, nextKey(prefix))
	if err != nil {
		panic(err)
	}
	defer it.Close()
	return it.Valid()
}
```

Reply:
Fixed